### PR TITLE
Fix accessing containers and image locks

### PR DESCRIPTION
**Pull request by rhatdan**
Match containers and image lock to the new layer lock method. in https://github.com/containers/storage/pull/1351

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
@mtrmac PTAL

File: containers.go [hunk 1]
```diff
@@ -265,6 +263,8 @@ func newContainerStore(dir string) (ContainerStore, error) {
 		bylayer:    make(map[string]*Container),
 		byname:     make(map[string]*Container),
 	}
+	cstore.Lock()
+	defer cstore.Unlock()
 	if err := cstore.Load(); err != nil {
```
Comment (vrothberg) [inline containers.go:268]: It's somehow surprising that callers of a public API are responsible for locking.

**Comment by rhatdan** [comment]
@vrothberg I think @mtrmac has a PR to make all of these functions Private to containers/storage.

**Comment by nalind** [comment]
LGTM

File: containers.go (hunk 1 above)
Comment (mtrmac) [reply containers.go:268]: The `{container,image,layer}Store` objects implement public interfaces, but AFAICS there is no public way to access them (one would have to take a `storage.Store` and manually cast it to an interface that contains accessors to the three stores).

Compare #1345 .

I think the basic idea is that various `storage.Store` operations need to coordinate _multiple_ operations on a single `layerStore`, so they need a consistent view, and a lock held across the `layerStore` calls.

See #1346 which tries to make the locking requirements much more explicit to make sure there are no mistakes there.

**Comment by mtrmac** [comment]
@rhatdan thanks!

**Pull request merged**